doc/sources/tests.rst

```diff
@@ -0,0 +1,221 @@
+
+.. code-block:: bash
+
+    pip install -r requirements-test.txt
+
+Some tests will only execute depending on the availability of optional dependencies at runtime. Other optional dependencies that will trigger additional tests can be installed as follows, assuming a Linux* system:
+
+.. code-block:: bash
+
+    pip install \
+        dpctl `# for GPU functionalities` \
+        dpnp `# for array API and GPU functionalities`
+
+    pip install --index-url https://software.repos.intel.com/python/pypi \
+        torch `# for array API`
+
+    pip install --index-url https://software.repos.intel.com/python/pypi \
+        mpi4py impi_rt `# for distributed mode, be sure to install from Intel's index` \
+        && pip install pytest-mpi `# also required, but not from Intel's index`
+
+.. warning:: It might not be possible to install all of the test dependencies simultaneously in the same Python environment. In particular, dependencies ``torch`` and ``dpctl`` / ``dpnp`` are likely to not be installable in a compatible way in the same environment if using pre-built distributions. Try using different Python environments for each set of dependencies to test.
+
+.. warning:: If installing dependencies for distributed mode from ``pip``, be sure to install ``mpi4py`` from the Intel ``pip`` index to ensure that it uses a compatible MPI backend. See :doc:`distributed-mode` for details.
+
+Executing tests
+~~~~~~~~~~~~~~~
+
+In order to run the whole test suite, the following script can be used on Linux*:
+
+.. code-block:: bash
+
+    conda-recipe/run_test.sh
+
+.. warning:: This script must be executed from the root of the repository.
+
+It also comes with an analog for Windows*:
+
+.. code-block:: console
+
+    call conda-recipe/run_test.bat
+
+Individual test files or tests can be executed with PyTest under different options (such as a different verbosity mode, stopping at the first failure, etc.) with calls to pytest - for example:
+
+.. code-block:: bash
+
+    pytest sklearnex/ensemble/tests/test_forest.py
+
+.. hint:: If executing these from the root of the repository, there might be naming clashes between the folders and the installed Python modules. It might be helpful to :ref:`build the library extensions in-place <build_script>` and set ``$PYTHONPATH`` to avoid problems.
+
+Configurable options
+~~~~~~~~~~~~~~~~~~~~
+
+The files ``run_test.sh`` and ``run_test.bat`` offer configurable behaviors through environment variables and command line arguments:
+
+- Environment variable ``$NO_DIST``, if set, will make it avoid running distributed mode tests. Note that executing these tests requires additional dependencies, otherwise they will be skipped either way.
```

Comment:
```suggestion
- Environment variable ``$NO_DIST``, if set, will avoid running distributed mode tests. Note that executing these tests requires additional dependencies, otherwise they will be skipped either way.
```